torchdata/dataloader2/dataloader2.py

```diff
@@ -281,3 +338,17 @@ def load_state_dict(self, state_dict: Dict[str, Any]) -> None:
             for adapter_fn in self.datapipe_adapter_fns:
                 self.datapipe = adapter_fn(self.datapipe)
         self._datapipe_before_reading_service_adapt = clone(self.datapipe)
+
+    def _pause(self):
+        self._is_paused = True
+        if hasattr(self.reading_service, "_pause"):
+            self.reading_service._pause()
+        else:
+            warnings.warn("ReadingService doesn't support pause.")
+
+    def _resume(self):
```

Comment:
Add a warning here when `_resume` is called when it is not paused, and then do no-op